proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpRequestHelper.java

```diff
@@ -0,0 +1,363 @@
+            // At first, try to deserialize JSON
+            final JsonObject jsonRoot = serialization.parseResponseObject(objectData);
+
+            // Check "status" property in received JSON
+            final JsonElement status = jsonRoot.get("status");
+            if (status == null || !status.isJsonPrimitive() || !status.getAsString().equalsIgnoreCase("OK")) {
+                throw buildResponseException(responseCode, responseData, jsonRoot, null);
+            }
+            if (endpoint.getResponseType() != null) {
+                // Check "responseObject" property, but only if response type is specified
+                final JsonElement responseObjectElement = jsonRoot.get("responseObject");
+                if (responseObjectElement != null && responseObjectElement.isJsonObject()) {
+                    // Now finally, try to deserialize the response object
+                    result = serialization.getGson().fromJson(responseObjectElement, endpoint.getResponseType().getType());
+                } else {
+                    result = null;
+                }
+                // Check if object was created and if not, then throw an exception.
+                if (result == null) {
+                    throw new JsonParseException("Failed to deserialize response object.");
+                }
+            } else {
+                // No action is needed. Response type is not specified, so the result can be null.
+                result = null;
+            }
+
+        } else {
+            // Response object is not wrapped in ObjectResponse<T>
+            if (endpoint.getResponseType() != null) {
+                // If response type is specified, then deserialize the object
+                result = serialization.deserializeObject(objectData, endpoint.getResponseType());
+
+            } else {
+                // No action is needed. If response type is not specified, then the result can be null.
+                result = null;
+            }
+        }
+
+        return result;
+    }
+
+
+    /**
+     * Constructs a {@link ErrorResponseApiException} or {@link FailedApiException} exceptions, depending
+     * on data received from the server. The method is package-private.
+     *
+     * @param responseCode HTTP response code
+     * @param responseData Response bytes
+     * @param jsonRoot Optional JSON representation. If object is not available, then the function will try to parse
+     * @param exception Optional exception. If provided, then its message will be used as a fallback.
+     * @return {@link Throwable} object with an appropriate exception.
+     */
+    @NonNull
+    public Throwable buildResponseException(int responseCode, @Nullable byte[] responseData, @Nullable JsonObject jsonRoot, @Nullable Throwable exception) {
+
+        // Convert bytes into String
+        final String responseString;
+        if (responseData != null) {
+            responseString = new String(responseData, Charset.defaultCharset());
```

Comment:
I think it is better to use `StandardCharsets.UTF_8` instead of `Charset.defaultCharset()`. The default charset can be changed externally, so I think it is better to stick to UTF-8.

Reply:
That's not possible on Android. See [documentation](https://developer.android.com/reference/java/nio/charset/Charset#defaultCharset())

Reply:
Ah, ok, ignore this comment then.